### DISCUSSION TITLE
Suggestion: Change the policy for `issues` (move Q&A to Discussions)

### DISCUSSION THREAD

**Discussion by ccrvlh** [fastapi/fastapi#5522] [Questions] [answered]
In my view, both for users and contributors in general, having a significant amount of issues can be challenging:

- For people looking for solutions to potential issues/bugs the large number of issues makes search less effective;
- For contributors or people who want to contribute, having the notifications on tends to not be friendly (lots of notifications for Q&A style issues);
- For outsiders that assess FastAPI as an option for a greenfield project, seeing not only a large number of issues open (+1k) and a large number of open/close ratio (somewhat close to 50% now) may give the impression (not discussing the merit of whether this is true or not) that the project may be abandoned or not properly maintained;

So far, what I could tell about the policy regarding issues is:
- If an issue is considered a question is tagged as "question", and left open;
- Issues without interaction for a long time are still opened;
- Issues with Q&A style with possible answers are kept opened, awaiting for the user who opened the issue to close it;
- Issues with answers are marked as "answered" and kept opened;
- Issues that are more closely related to discussions are kept opened for interactions (discussions on translations for example);

My impression is that this policy can be detrimental to the project as a whole, and negatively contribute to outsider's perception on the project.

I would like to suggest for Q&A style issues to be immediately closed and/or moved to the Discussions feature with a Q&A tag, and make issues strictly dedicated to: (1) bugs and (2) feature requests that are specific and small enough so that they can be assessed rather quickly (this is to avoid feature requests that demand lot's of discussions about design, implementation and other details).

Even though the issues template help in this sense, the policy for older (and not necessarily old) issues still favors opened issues, a more loose policy. I suggest to have a stricter policy for opened issues, and have a flexible policy for discussions.

In short, actionable suggestions to what I think could improve the project:
- Implement a bot to close stale issues (make an aggressive policy at first, and more lose afterwards);
- Immediately move all Q&A style issues ("How do I ...") to the Discussions/Q&A tab;
- Immediately move all translation related work in progress to the Discussions tab;
- Close issues that have answers (close first, give the option to the user to open it again);

Part of effects and potential solutions to what I'm considering a challenge here, were also [discussed here](https://github.com/tiangolo/fastapi/discussions/3970#discussioncomment-2530905).

I'm happy to contribute.

BIG DISCLAIMER: This is a suggestion, and mostly based on a personal perception. I understand if other people don't share the same perception as I do.

**Comment by tiangolo** [comment, marked as answer]
Thanks for the input! And yes, I've been wanting to move questions to discussions since I was allowed in the beta.

I compute FastAPI Experts from the answers to others. They released an API to access discussions, so that would probably enable that part. But the template and more strict form for creating an issue is still not supported by discussions, that's what would save most is the time, and that's why I haven't moved to that yet.

Also, the label "answered" is used by my GitHub action issue-manager to close issues after being answered, not having a response from others, and a period of inactivity.

But I have to add that label in the right places.

For translations, it makes sense to have it in discussions too, but I have to update all the setup for that, the GitHub actions that notify about new translations to review, etc.

**Reply by ccrvlh** [reply]
As a benchmark, some relevant projects (SQLAlchemy I think is a great example) have a more loose policy on the Discussions area, so questions have more of a free form, instead of the heavy template-based style for issues. Since Q&As can be mostly community based, and no permissions are necessary, moving Q&A to discussions may be positive even in the scenario where there are no templates. Anyways, understood the approach. Is does seem intensive in manual work, but gives more control over the lifecycle of issues, I guess this is probably the most relevant tradeoff.

**Reply by tiangolo** [reply]
The thing is that more free form questions mean a lot more effort and back and forth from the people answering and helping, trying to get all the information.

There's normally a small bunch of people helping others and a large bunch of people asking questions. It's not even common that those who answer also ask questions. And I don't want to require more work from the few that come to help, trying to get the information, that's the main reason... 😅

**Reply by tiangolo** [reply]
Now GitHub Discussions finally supports templates, so I moved the questions from issues to Discussions. :tada:
